filedescriptor/src/lib.rs

```diff
@@ -257,6 +264,22 @@ impl FileDescriptor {
     pub fn set_non_blocking(&mut self, non_blocking: bool) -> anyhow::Result<()> {
         self.set_non_blocking_impl(non_blocking)
     }
+
+    /// Attempt to redirect stdio to the underlying handle and return
+    /// a `FileDescriptor` wrapped around the original stdio source.
+    /// Since the redirection requires kernel resources that may not be
+    /// available, this is a potentially fallible operation.
+    /// Supports StdIn, StdOut, and StdErr redirections.
```

Comment:
nit: the case of these doesn't match those in the enum.  I'd be fine omitting this line TBH, but if we keep it, I'd suggest making this consistent! :)

Reply:
Sure, makes sense :) will make these all lowercase then.

Reply:
Done